rosco-core/src/test/groovy/com/netflix/spinnaker/rosco/executor/BakePollerSpec.groovy

```diff
@@ -129,4 +136,72 @@ class BakePollerSpec extends Specification {
       1 * bakeStoreMock.retrieveBakeStatusById(JOB_ID) >> new BakeStatus()
   }
 
+  void 'Decorate the bakeDetails with an artifact if bake is successfull'() {
+    setup:
+      def cloudProviderBakeHandlerRegistryMock = Mock(CloudProviderBakeHandlerRegistry)
+      def cloudProviderBakeHandlerMock = Mock(CloudProviderBakeHandler)
+      def bakeStoreMock = Mock(RedisBackedBakeStore)
+      def jobExecutorMock = Mock(JobExecutor)
+      def bakeRequest = new BakeRequest(build_info_url: SOME_BUILD_INFO_URL)
+      def bakeRecipe = new BakeRecipe(name: SOME_BAKE_RECIPE_NAME, version: SOME_APP_VERSION_STR, command: [])
+      def bakedArtifact = new Artifact(
+        name: bakeRecipe.name,
+        version: bakeRecipe.version,
+        type: DOCKER_CLOUD_PROVIDER,
+        reference: AMI_ID,
+        metadata: [
+          build_info_url: bakeRequest.build_info_url,
+          build_number: bakeRequest.build_number
+        ]
+      )
+      def bakeDetails = new Bake(id: JOB_ID, ami: AMI_ID, image_name: IMAGE_NAME, artifact: bakedArtifact)
+
+      @Subject
+      def bakePoller = new BakePoller(
+        bakeStore: bakeStoreMock,
+        executor: jobExecutorMock,
+        cloudProviderBakeHandlerRegistry: cloudProviderBakeHandlerRegistryMock,
+        registry: new DefaultRegistry())
+
+    when:
+      bakePoller.completeBake(JOB_ID, LOGS_CONTENT)
+
+    then:
+      1 * bakeStoreMock.retrieveCloudProviderById(JOB_ID) >> DOCKER_CLOUD_PROVIDER.toString()
+      1 * cloudProviderBakeHandlerRegistryMock.lookup(DOCKER_CLOUD_PROVIDER) >> cloudProviderBakeHandlerMock
+      1 * bakeStoreMock.retrieveRegionById(JOB_ID) >> REGION
+      1 * cloudProviderBakeHandlerMock.scrapeCompletedBakeResults(REGION, JOB_ID, LOGS_CONTENT) >> bakeDetails
+      1 * bakeStoreMock.retrieveBakeRequestById(JOB_ID) >> bakeRequest
+      1 * bakeStoreMock.retrieveBakeRecipeById(JOB_ID) >> bakeRecipe
+      1 * bakeStoreMock.updateBakeDetails(bakeDetails)
+  }
+
+  void 'Skip artifact decoration if required data is missing'() {
```

Comment:
Yes you're right. I struggled to get this test right, which made me pull out `produceArtifactDecorationFrom` from BakePoller, so that it was a simple contained method, which would be easier to test. I ment to delete this test and replace it with more accurate testing in `CloudProviderBakeHandlerSpec`. I'll clean this up.